### DISCUSSION TITLE
Custom added icons are not visible anymore

### DISCUSSION THREAD

**Discussion by alx-xlx** [dani-garcia/vaultwarden#5276] [Q&A] [answered]
I added few custom icons for some selfhosted websites two months ago into the `icon_cache` directory. and icons were correctly displaying. However I now realize they are not displaying anymore.

![](https://i.imgur.com/MgokrPx.png)

![](https://i.imgur.com/FLWmmLV.png)

**Comment by BlackDex** [comment]
Is there a .miss file also for that icon? If so, remove that.

**Reply by alx-xlx** [reply]
I did `sudo rm -rf *.miss` on the `icon_cache` dir

and restarted the container, but that didn't solve the issue.

I checked the docker logs and I see this

```
[2024-12-12 04:15:37.749][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2024-12-12 04:15:39.265][request][INFO] GET /icons/uptime.example.org/icon.png
[2024-12-12 04:15:39.267][vaultwarden::api::icons][WARN] IP 192.168.0.113 for domain 'uptime.example.org' is not a global IP!
[2024-12-12 04:15:39.267][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2024-12-12 04:15:39.267][request][INFO] GET /icons/uptimerobot.com/icon.png
[2024-12-12 04:15:39.268][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
```

**Comment by alx-xlx** [comment]
this is the PNG file that I am using

![uptime example org](https://github.com/user-attachments/assets/491ee64f-53a9-4cda-bd65-2cbf5446e6bd)

**Comment by BlackDex** [comment]
Also, those are cached. So it might be still cache of the browser which is returning the same image.

**Reply by alx-xlx** [reply]
I tried on a different browser, but I get same behaviour.
Which location has the icon saved ?

**Reply by BlackDex** [reply]
The icons are located in `/data/icons`  where `/data` is the data folder you defined your self. 

Also check for `.miss` files as i mentioned before.

**Reply by stefan0xC** [reply]
@BlackDex isn't the issue that those icons got deleted because they (presumably) expired?
https://github.com/dani-garcia/vaultwarden/blob/45e5f06b86c66d30ece0cf9169d3a5b65a87e380/.env.template#L224-L225

Or if they are not deleted they would still not be served because they have expired:
https://github.com/dani-garcia/vaultwarden/blob/45e5f06b86c66d30ece0cf9169d3a5b65a87e380/src/api/icons.rs#L245-L246

**Reply by BlackDex** [reply]
Well, the do not get deleted. But a `.miss` file will be created.
One way to fix that is just running `touch my.icon.tld.png`

**Reply by stefan0xC** [reply]
> One way to fix that is just running `touch my.icon.tld.png`

So the solution would be to periodically update the last modification date for custom icons, if you don't want to set `ICON_CACHE_TTL=0` (which is more for when you disable icon downloads altogether). Should we update the documentation somewhere?

**Reply by alx-xlx** [reply]
> The icons are located in `/data/icons` where `/data` is the data folder you defined your self.
> 
> Also check for `.miss` files as i mentioned before.


I don't have the `icons` dir

![q4ZC2XNoWe](https://github.com/user-attachments/assets/70ac2b98-c0fd-4690-86fe-3d03abce77d0)

![IAxmx7jGGx](https://github.com/user-attachments/assets/8a2f14bf-ca4b-47fa-b6b7-21934027c259)




This is my compose file

```
docker run -d --name vaultwarden \
  -v /home/pi/data/vaultwarden/:/data/ \
  -p 4179:80 \
  --label sablier.enable=true \
  --label sablier.group=vaultwarden \
  --health-cmd="curl -f http://localhost:80 || exit 1" \
  --health-interval=30s \
  --health-timeout=5s \
  --health-retries=3 \
  --health-start-period=10s \
  vaultwarden/server:latest

```

**Reply by BlackDex** [reply]
`icon_cache`

**Reply by alx-xlx** [reply]
> `icon_cache`

its pretty weird because custom icons were working 2months ago, I didn't make any changes to vaultwarden except updating it to the latest version using watchtower

there is some issue in the [logs](https://github.com/dani-garcia/vaultwarden/discussions/5276#discussioncomment-11539990)

**Reply by BlackDex** [reply]
It isn't weird. If the icon last modified time is past the caching lifetime, Vaultwarden tries to fetch a new one. If that doesn't work, it currently marks it as stale and returns the globe icon.

**Reply by BlackDex** [reply, marked as answer]
Also, it doesn't allow internal IP's by default, you can disable this. by modifying this setting:
https://github.com/dani-garcia/vaultwarden/blob/fd512300448995735961cbeedcbe4b684be1c5fd/.env.template#L341

**Reply by alx-xlx** [reply]
> Also, it doesn't allow internal IP's by default, you can disable this. by modifying this setting:
> 
> https://github.com/dani-garcia/vaultwarden/blob/fd512300448995735961cbeedcbe4b684be1c5fd/.env.template#L341

this fixed the issue, thanks everyone for your guidance
